joblib/_compat.py

```diff
@@ -17,3 +17,11 @@
 def with_metaclass(meta, *bases):
     """Create a base class with a metaclass."""
     return meta("NewBase", bases, {})
+
+
+# python2.7 error compatibility
+if PY27:
+    class CompatFileExistsError(OSError):
+        pass
+else:
+    CompatFileExistsError = FileExistsError
```

Comment:
Another way to do it:

```python

try:
    FileExistsError
except NameError:
    # compat for Python 2.7
    FileExistsError = OSError
```

but I am fine with both solutions

Reply:
maybe using an explicit different name `CompatFileExistsError` is better to make it easy to find where this is actually needed/used in the code base.

Reply:
You cannot import `FileExistsError` from the compat module I think. That is why I had to resort to this compat name. Also, as you said, it might be better to be explicit here.